examples/fcis/train_multi.py

```diff
@@ -0,0 +1,154 @@
+from __future__ import division
+
+import argparse
+import numpy as np
+
+import chainer
+from chainer.optimizers.momentum_sgd import MomentumSGDRule
+from chainer.training import extensions
+from chainer.training.triggers import ManualScheduleTrigger
+import chainermn
+
+from chainercv.chainer_experimental.datasets.sliceable \
+    import TransformDataset
+from chainercv.datasets import sbd_instance_segmentation_label_names
+from chainercv.datasets import SBDInstanceSegmentationDataset
+from chainercv.experimental.links import FCISResNet101
+from chainercv.experimental.links import FCISTrainChain
+from chainercv.extensions import InstanceSegmentationVOCEvaluator
+
+from train import concat_examples
+from train import Transform
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='ChainerCV training example: FCIS')
+    parser.add_argument('--out', '-o', default='result',
+                        help='Output directory')
+    parser.add_argument('--seed', '-s', type=int, default=0)
+    parser.add_argument('--lr', '-l', type=float, default=0.0005)
+    parser.add_argument(
+        '--lr-cooldown-factor', '-lcf', type=float, default=0.1)
+    parser.add_argument('--epoch', '-e', type=int, default=44)
+    parser.add_argument('--cooldown-epoch', '-ce', type=int, default=28)
+    args = parser.parse_args()
+
+    # chainermn
+    comm = chainermn.create_communicator()
+    device = comm.intra_rank
+
+    np.random.seed(args.seed)
+
+    # model
+    fcis = FCISResNet101(
+        n_fg_class=len(sbd_instance_segmentation_label_names),
+        pretrained_model='imagenet', iter2=False)
+    fcis.use_preset('evaluate')
+    model = FCISTrainChain(fcis)
+    chainer.cuda.get_device_from_id(device).use()
+    model.to_gpu()
+
+    # dataset
+    train_dataset = TransformDataset(
+        SBDInstanceSegmentationDataset(split='train'),
+        ('img', 'mask', 'label', 'bbox', 'scale'),
+        Transform(model.fcis))
+    if comm.rank == 0:
+        indices = np.arange(len(train_dataset))
+    else:
+        indices = None
+    indices = chainermn.scatter_dataset(indices, comm, shuffle=True)
+    train_dataset = train_dataset.slice[indices]
+    train_iter = chainer.iterators.SerialIterator(train_dataset, batch_size=1)
+
+    if comm.rank == 0:
+        test_dataset = SBDInstanceSegmentationDataset(split='val')
+        test_iter = chainer.iterators.SerialIterator(
+            test_dataset, batch_size=1, repeat=False, shuffle=False)
+
+    # optimizer
+    optimizer = chainermn.create_multi_node_optimizer(
+        chainer.optimizers.MomentumSGD(lr=args.lr * comm.size, momentum=0.9),
```

Comment:
`args.lr * comm.size --> args.lr`

The actual `lr` should be equal to what is passed from the command line.
It gets harder to debug when linear scaling rule is implicitly encoded in a training script.

Reply:
is there any solution to stop gradient mean in `ChainerMN`?

Reply:
What do you mean by `stop gradient mean`?

Reply:
chainermn default is gather gradient, average it and backpropagate it to all model. 
i want to just skip the mean phase.

Reply:
Unfortunately, I don't know how to that.

I like to keep loss divided by batchsize always, so I never tried doing it.

Reply:
I updated and add help.